tutorial.md

```diff
@@ -0,0 +1,1022 @@
+  exist yet.  Instead of it, use `xmobar` and carefully read the part
+  about pipes later on (`xmobar` uses pipes to make xmobar talk to
+  xmonad).
+
+Notice how `$` became `.`!  The dot operator `(.)` in Haskell means
+function composition and is read from right to left.  What this means in
+this specific case is essentially the following:
+
+  > take the three functions `xmonad`, `ewmhFullscreen`, and `ewmh` and
+  > give me the big new function `xmonad . ewmhFullscreen . ewmh` that
+  > first executes `ewmh`, then `ewmhFullscreen`, and finally `xmonad`.
+  > Then give it `xmobarProp myConfig` as its argument so it can do its
+  > thing.
+
+This should strike you as nothing more than a syntactical quirk.  If you
+want, you can also write this using Haskell's `do`-notation and our good
+old dollar sign (this may feel more natural if you already have
+programming experience in an imperative language):
+
+``` haskell
+  main :: IO ()
+  main = do
+      xmobarProcess <- xmobarProp myConfig
+      xmonad $ ewmhFullscreen $ ewmh $ xmobarProcess
+```
+
+Back to the meaning of the code.  What it does is take our tweaked
+default configuration (`myConfig`) and add the support we need to make
+xmobar our status bar.  Do note that you will also need to add the
+`XMonadLog` plugin to your xmobar configuration; we will do this
+together below, so don't sweat it for now.
+
+To understand why this is necessary, let's talk a little bit about how
+xmonad and xmobar fit together.  You can piece them together in several
+different ways.
+
+By default, xmobar accepts input on its stdin, which it can display at
+an arbitrary position on the screen.  We want xmonad to send xmobar the
+stuff that I have at the upper left of the starting screenshots:
+information about available workspaces, current layout, and open
+windows.  Naively, we can achieve this by spawning a pipe and letting
+xmonad feed the relevant information to that pipe.  The problem with
+that approach is that when the pipe is not being read and gets full,
+xmonad will freeze!
+
+It is thus much better to switch over to property based logging, where
+we are writing to an X11 property and having xmobar read that; no danger
+when things are not being read!  For this reason we have to use
+`XMonadLog` instead of `StdinReader` in our xmobar.  There's also an
+`UnsafeXMonadLog` available, should you want to send actions to xmobar
+sometimes (this is useful, for example, for
+[XMonad.Util.ClickableWorkspaces], which is a new feature in `0.17`).
+
+_IF YOU ARE ON A VERSION `< 0.17`_: As discussed above, the `xmobar`
+  function uses pipes, so you actually do want to use the `StdinReader`.
+  Simply replace _all_ occurences of `XMonadLog` with `StdinReader`
+  below (don't forget the template!)
+
+## Configuring Xmobar
+
+Now, before this will work, we have to configure xmobar.  Here's a nice
+starting point.
+
+``` haskell
+  Config { overrideRedirect = False
```

Comment:
It may be worth noting that this config file only _looks_ like Haskell code. I've seen people try to use e.g. expressions or bindings in it.